src/bufio.c

```diff
@@ -981,6 +980,8 @@ error has occured and the status code of the stream was set.
 
   BUFIO_TIMEDOUT A write operation or poll timed out.
 
+  BUFIO_EOF      Reached end-of-file.
```

Comment:
EOF is normally not used for write operations.

I would tend to introduce a new status code similar to `ENOSPC`. We would need to properly check/set that for file writes as well (e.g., combining EDQUOT and ENOSPC) - which needs a new test. @_@

What do you think?

Reply:
Fully agree on the principle. In this particular case `ENOBUFS` would be even more fitting, judging from the description and other usage. or `EMSGSIZE`.
`ENOSPC` could represent those as well with a broader interpretation of what a `device` is. :)